app.rb

```diff
@@ -24,9 +24,12 @@ class App < Sinatra::Base
           siteIP: params.fetch(:site_ip_address),
           building_identifier: params.fetch(:called_station_id),
         )
+
         user = User.find(username: params.fetch(:username))
-        user.last_login = Time.now.strftime('%y-%m-%d %H:%M:%S')
-        user.save
+        if user
+          user.last_login = Time.now.strftime('%y-%m-%d %H:%M:%S')
+          user.save
+        end
```

Comment:
Looks like a solid candidate for an use case 🎉

Reply:
Yeah could be.  I have yet to refactor this code out of app.rb

Reply:
Yeah, would be nice to see some refactoring in a later PR.